providers/base/tests/test_wifi_nmcli_test.py

```diff
@@ -36,3 +51,563 @@ def test_legacy_nmcli_false(self, subprocess_mock):
+        sp_call_mock.assert_called_with("nmcli c up uuid2".split())
+
+    @patch("wifi_nmcli_test.sp.call", side_effect=Exception("Command failed"))
+    @patch("wifi_nmcli_test.get_nm_activate_connection", return_value="")
+    def test_connection_activation_fails_due_to_exception(
+        self,
+        get_nm_activate_connection_mock,
+        sp_call_mock,
+    ):
+        turn_up_connection("uuid3")
+
+
+class TestTurnDownNmConnections(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.call")
+    @patch("wifi_nmcli_test._get_nm_wireless_connections", return_value={})
+    def test_no_connections_to_turn_down(
+        self, get_connections_mock, sp_call_mock
+    ):
+        turn_down_nm_connections()
+        self.assertEqual(get_connections_mock.call_count, 1)
+        sp_call_mock.assert_not_called()
+
+    @patch("wifi_nmcli_test.sp.call")
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={"Wireless1": {"uuid": "uuid1", "state": "activated"}},
+    )
+    def test_turn_down_single_connection(
+        self, get_connections_mock, sp_call_mock
+    ):
+        turn_down_nm_connections()
+        self.assertEqual(get_connections_mock.call_count, 1)
+        sp_call_mock.assert_called_once_with("nmcli c down uuid1".split())
+
+    @patch(
+        "wifi_nmcli_test.sp.call", side_effect=Exception("Error turning down")
+    )
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={"Wireless1": {"uuid": "uuid1", "state": "activated"}},
+    )
+    def test_turn_down_single_connection_with_exception(
+        self, get_connections_mock, sp_call_mock
+    ):
+        turn_down_nm_connections()
+        self.assertEqual(get_connections_mock.call_count, 1)
+        sp_call_mock.assert_called_once_with("nmcli c down uuid1".split())
+
+    @patch("wifi_nmcli_test.sp.call")
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={
+            "Wireless1": {"uuid": "uuid1", "state": "activated"},
+            "Wireless2": {"uuid": "uuid2", "state": "activated"},
+        },
+    )
+    def test_turn_down_multiple_connections(
+        self, get_connections_mock, sp_call_mock
+    ):
+        turn_down_nm_connections()
+        self.assertEqual(get_connections_mock.call_count, 1)
+        calls = [
+            call("nmcli c down uuid1".split()),
+            call("nmcli c down uuid2".split()),
+        ]
+        sp_call_mock.assert_has_calls(calls, any_order=True)
+
+
+class TestDeleteTestApSsidConnection(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.call", return_value=0)
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={
+            "TEST_CON": {"uuid": "uuid-test", "state": "deactivated"}
+        },
+    )
+    @patch("wifi_nmcli_test.print")
+    def test_delete_existing_test_con(
+        self, get_nm_wireless_connections_mock, sp_call_mock, print_mock
```

Comment:
The order of the mocks is wrong, it's inverted when mocking. Check this because it is happening many times during the tests
```suggestion
        self, print_mock, get_nm_wireless_connections_mock, sp_call_mock, 
```

Reply:
Corrected the order